### PR TITLE
Use endpoint IDs for endpoint predict requests

### PR DESCRIPTION
**Pull request by phil-scale**
Extra latency right now for sync/async requests incurred from querying for endpoint by name

### CODE REVIEW

File: launch/client.py [hunk 1]
```diff
@@ -1187,7 +1185,6 @@ def _async_request(
                 `abcabcab-cabc-abca-0123456789ab`
         """
         validate_task_request(url=url, args=args)
-        endpoint = self.get_model_endpoint(endpoint_name)
```
Comment (yunfeng-scale) [inline launch/client.py:1190]: is this just a postgres call?

**Review by yunfeng-scale — APPROVED** [review]

**Review by yixu34 — COMMENTED** [review]
Wait is this a breaking change?

File: launch/client.py (hunk 1 above)
Comment (phil-scale) [reply launch/client.py:1190]: Yes, but also Redis to get some cached data about the infra state (and k8s api server if Redis cache doesn't contain the info)

**Comment by phil-scale** [comment]
> Wait is this a breaking change?

Sort of? `_async_request` and `_sync_request` were _technically_ deprecated and private anyways

**Comment by yixu34** [comment]
> > Wait is this a breaking change?
> 
> Sort of? `_async_request` and `_sync_request` were _technically_ deprecated and private anyways

True, though I think enough people might have latched onto `_async_request`. Maybe we can limit the change to just `_sync_request` for now?

Maybe we can do a broader change as part of #73?

**Comment by phil-scale** [comment]
Ok closing this in favor of modifying https://github.com/scaleapi/launch-python-client/pull/73 instead

**Pull request closed without merging**